core/rendered_connection.ts

```diff
@@ -644,6 +645,14 @@ export class RenderedConnection
   /** See IFocusableNode.onNodeFocus. */
   onNodeFocus(): void {
     this.highlight();
+    const highlight = this.findHighlightSvg();
+    if (!highlight) return;
+    const bbox = highlight.getBBox();
+    // Y coordinate appears to be the middle of the connection.
```

Comment:
```suggestion
    // Y coordinate is in the middle of the connection.
```

Otherwise this reads as a discovery rather than a description.

Reply:
It kind of is a discovery; I can't find anywhere that this is documented, but experimentally it checked out. I'm hesitant to state it as a fact.

Reply:
I worry that this will make it more confusing when trying to reason about this code in the future.

I dug into this a bit, too, and it surprises me. I think bbox is supposed to be providing an upper left position, not centered (though I have seen some weird bbox behaviors in SVGs outside of Blockly). If I understand the documentation correctly, though, bbox isn't affected by transformations which might be skewing things here).

I wonder: since the highlight SVG is just a clone of the block's path with some small adjustments, would `svg_math`'s `getRelativeXY` work for this? That might be a nicer approach if so since then it'll at least be consistent with blocks.

Reply:
I concur; converting to using the source block's position plus the relative XY of the highlight path yielded the same results as the bounding box, such that navigating backwards to an open input connection ~mostly scrolls it into view, but not along the vertical axis:

<img width="372" height="305" alt="Screenshot 2025-08-15 at 1 12 09 PM" src="https://github.com/user-attachments/assets/4e9248e1-6eb1-49f8-b6ce-59d2a46c8524" />

I suppose I'm happy that they're the same, but I was also hoping that would resolve this. Another possibility might be to just scroll the source block itself into view?

Reply:
So something weird is happening here, then, with the highlight SVG. If it's possible to scroll the source block into view, that probably does sound cleaner and more consistent to how we'd solve other similar "part of the block is off screen" problems.

Reply:
Done.